tekton/api/golang-db-test.yaml

```diff
@@ -0,0 +1,81 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: golang-db-test
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: test
+    tekton.dev/displayName: "golang test"
+spec:
+  description: >-
+    This Task is Golang task to test Go projects.
+
+  params:
+    - name: package
+      description: package (and its children) under test
+    - name: packages
+      description: "packages to test (default: ./...)"
+      default: "./..."
+    - name: path_context
+      description: path to the directory to use as context.
+      default: "."
+    - name: version
+      description: golang version to use for tests
+      default: "latest"
+    - name: flags
+      description: flags to use for the test command
+      default: -race -cover -v
+    - name: GOOS
+      description: "running program's operating system target"
+      default: linux
+    - name: GOARCH
+      description: "running program's architecture target"
+      default: amd64
+    - name: GO111MODULE
+      description: "value of module support"
+      default: auto
+  workspaces:
+    - name: source
+  steps:
+    - name: unit-test
+      image: docker.io/library/golang:$(params.version)
+      script: |
+        SRC_PATH="$GOPATH/src/$(params.package)/$(params.path_context)"
+        mkdir -p $SRC_PATH
+        cp -R "$(workspaces.source.path)"/"$(params.path_context)"/* $SRC_PATH
+        cd $SRC_PATH
+        go test $(params.flags) $(params.packages)
+      env:
+        - name: GOOS
+          value: "$(params.GOOS)"
+        - name: GOARCH
+          value: "$(params.GOARCH)"
+        - name: GO111MODULE
+          value: "$(params.GO111MODULE)"
+        - name: POSTGRES_HOST
+          value: localhost
+        - name: POSTGRES_DB
+          value: hub_test
+        - name: POSTGRES_USER
+          value: postgres
+        - name: POSTGRES_PASSWORD
+          value: postgres
+        - name: POSTGRES_PORT
+          value: "5432"
+  sidecars:
+    - image: postgres:12.5
```

Comment:
I am wondering if we should be tagging the image here with the digest so that our CI doesn't breaks. WDYT?

Reply:
yup. will use the same image as in the deployment